File: my-folder/0078-subsets/solution.go
func subsets(nums []int) [][]int {
    var res [][]int

    var subset []int

    var dfs func(int)
    dfs = func(i int) {
        if i >= len(nums) {
            tmp := make([]int, len(subset))
            copy(tmp, subset)
            res = append(res, tmp)
            return
        }
        subset = append(subset, nums[i])
        dfs(i + 1)

        subset = subset[:len(subset)-1]
        dfs(i + 1)
    }

    dfs(0)

    return res

}

File: my-folder/0078-subsets/solution.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    let subset = []
    let result = []

    function backtrack(i) {
        if (i >= nums.length) {
            result.push([...subset])
            return
        }

        subset.push(nums[i])
        backtrack(i+1)

        subset.pop()
        backtrack(i+1)
    }

    backtrack(0)
    return result
};
